lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/Builtin.java

```diff
@@ -12,10 +43,75 @@
   /** @return A fully qualified name of the corresponding Enso type in standard library. */
   String stdlibName() default "";
 
+  /** @return A custom name of the builtin type * */
+  String name() default "";
+
+  /**
+   * A method (or constructor) annotated with {@link Builtin.Method} will generate an implementation
+   * for {@link BuiltinMethod} {@link com.oracle.truffle.api.nodes.Node} class.
+   *
+   * <p>In its simplest form, annotating a simple non-overloaded with the annotation will generate a
+   * concrete class with a single `execute` method, inside which we call the annotated method.
+   *
+   * <p>Consider the following class and its to-be-generated builtin methods
+   *
+   * <pre>
+   * public class Foo extends {@link TruffleObject} {
+   *    // ...
+   *    {@link Builtin.Method @Builtin.Method}
+   *    public long length() {
+   *      // ...
+   *    }
+   *
+   *    {@link Builtin.Method @Builtin.Method}
+   *    public static Foo empty(Object element) {
+   *        // ...
+   *    }
+   * }
+   * </pre>
+   *
+   * The annotation processor will generate the corresponding builtin classes
+   *
+   * <pre>
+   * {@link BuiltinMethod @BuiltinMethod}(type = "Foo", name = "length")
+   * public class LengthFooNode extends Node {
+   *
+   *   long execute(Array _this) {
```

Comment:
```suggestion
   *   long execute(Foo _this) {
```

I don't understand why the argument would be `Array`?

Reply:
c&p